vochain/transaction.go

```diff
@@ -185,54 +185,155 @@ func (app *BaseApplication) AddTx(vtx *VochainTx, commit bool) (*AddTxResponse,
 			return response, app.State.AddToRollingCensus(tx.ProcessId, tx.NewKey, weight)
 		}
 
-	case *models.Tx_SetAccountInfo:
-		txValues, err := SetAccountInfoTxCheck(vtx.Tx, vtx.SignedBody, vtx.Signature, app.State)
-		if err != nil {
-			return nil, fmt.Errorf("setAccountInfoTxCheck: %w", err)
+	case *models.Tx_SetAccount:
+		tx := vtx.Tx.GetSetAccount()
+		var err error
+		switch tx.Txtype {
+		case models.TxType_CREATE_ACCOUNT:
+			err = CreateAccountTxCheck(vtx.Tx, vtx.SignedBody, vtx.Signature, app.State)
+			if err != nil {
+				return nil, fmt.Errorf("createAccount: %w", err)
+			}
+		case models.TxType_SET_ACCOUNT_INFO_URI:
+			err = SetAccountInfoTxCheck(vtx.Tx, vtx.SignedBody, vtx.Signature, app.State)
+			if err != nil {
+				return nil, fmt.Errorf("setAccountInfoTxCheck: %w", err)
+			}
+		case models.TxType_ADD_DELEGATE_FOR_ACCOUNT, models.TxType_DEL_DELEGATE_FOR_ACCOUNT:
+			err = SetAccountDelegateTxCheck(vtx.Tx, vtx.SignedBody, vtx.Signature, app.State)
+			if err != nil {
+				return nil, fmt.Errorf("setAccountDelegateTxCheck: %w", err)
+			}
+		default:
+			return nil, fmt.Errorf("setAccount: invalid transaction type")
 		}
+
 		if commit {
-			tx := vtx.Tx.GetSetAccountInfo()
-			// create account
-			if txValues.CreateAccount {
-				// with faucet payload provided
-				if txValues.CreateAccountWithFaucet {
-					// create account
-					if err := app.State.CreateAccount(
-						txValues.TxSender,
-						tx.GetInfoURI(),
-						make([]common.Address, 0),
-						0,
-					); err != nil {
-						return nil, fmt.Errorf("setAccountInfoTxCheck: createAccount %w", err)
+			switch tx.Txtype {
+			case models.TxType_CREATE_ACCOUNT:
+				txSenderAddress, err := ethereum.AddrFromSignature(vtx.SignedBody, vtx.Signature)
+				if err != nil {
+					return nil, fmt.Errorf("createAccountTx: txSenderAddress %w", err)
+				}
+				delegates := make([][]byte, 0)
```

Comment:
Third time I see this code

Reply:
moved this check out of the commit phase, and added a generic checkDuplicateDelegates util function